src/AppInstallerCommonCore/Public/winget/ManifestYamlWriter.h

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+#include <winget/Manifest.h>
+#include <winget/Yaml.h>
+
+namespace AppInstaller::Manifest::YamlWriter
+{
+    /// <summary>
+    /// Converts the manifest and a single installer to a yaml string.
+    /// </summary>
+    /// <param name="manifest">Manifest object.</param>
+    /// <param name="installer">Manifest installer object.</param>
+    /// <returns>Yaml string.</returns>
+    std::string ManifestToYamlString(const Manifest& manifest, const ManifestInstaller& installer);
+
+    /// <summary>
+    /// Exports the manifest yaml string to a file.
+    /// </summary>
+    /// <param name="content">Manifest yaml string.</param>
+    /// <param name="out">Path of the yaml file.</param>
+    void OutputYamlFile(const std::string& content, const std::filesystem::path& out);
```

Comment:
This should be 
void OutputYamlFile(const Manifest& manifest, const ManifestInstaller& installer, const std::filesystem::path& out);

And set the yaml emitter to the file stream, so we don't need to hold a big string in memory. Then update the caller to use this method directly.

The above ManifestToYamlString could still be kept for future use. And both ManifestToYamlString and OutputYamlFile could use a common function which takes in the emitter

Reply:
Changed to suggested.